runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -4181,6 +4239,9 @@ void JitShutdown(J9JITConfig * jitConfig)
 
    TR::CompilationInfo * compInfo = TR::CompilationInfo::get(jitConfig);
 
+   if (jitConfig->serverAOTMethodSet)
```

Comment:
We need to destruct the set. Calling just jitPersistentFree will only deallocate the memory used for the set itself, but not the entries added to the set.